File: python/algorithms/easy/solveMeFirst.py
"""
Author: Rithie Natan Carvalhaes Prado
Date: 2026-04-09
Last modified: 2026-04-09
Description: Solve Me First - HackerRank
Test description: https://www.hackerrank.com/challenges/solve-me-first/problem
"""
#---------- functions ----------
def solveMeFirst(a: int, b: int) -> int:
    """
    It takes two integers as input and returns their sum.

    Parameters:
    a (int): The first integer.
    b (int): The second integer.

    Returns:
    int: The sum of the two integers.

    Raises:
    ValueError: If either a or b is negative, or if either a or b is greater than 1000.
    """
    if a <= 0 or b <= 0 or a > 1000 or b > 1000:
        raise ValueError("Input integers must be within the range of 1 to 1000.")

    return a + b


#---------- Main ----------
if __name__ == "__main__":
    """
    The main function to test other kind of inputs.
    """
    num1 = int(input())
    num2 = int(input())
    res = solveMeFirst(num1, num2)
    print(res)

File: typescript-javascript/algorithms/easy/solveMeFirst.ts
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-04-09
 * Last modified: 2026-06-11
 * @description Solve Me First - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/solve-me-first/problem}
 */
//---------- functions ----------
/**
 * Soolve Me First function that takes two integers and returns their sum.
 * 
 * @param {number} a
 * @param {number} b 
 * @returns {number} sum of a and b.
 * @throws {Error} If a or b are not within the range of 1 to 1000.
 */
function solveMeFirst(a: number, b: number): number
{ 
    if (a <= 0 || b <= 0 || a > 1000 || b > 1000)
    { throw new Error("Input values must be between 1 and 1000."); }

    return a + b; 
}//end solveMeFirst()